### PR TITLE
Add org.sugarlabs.Pippy

### PR DESCRIPTION
**Pull request by tchx84**
Pippy allows the student to examine, execute, and modify simple Python programs. It's possible to write Python statements to play sounds, calculate expressions, or make simple text based interactive games. It also includes PyGame examples and a tutorial for GTK basics.

### CODE REVIEW

**Comment by tchx84** [comment]
bot, build org.sugarlabs.Pippy

File: org.sugarlabs.Pippy.json [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.34",
+    "sdk": "org.gnome.Sdk",
+    "separate-locales": false,
+    "command": "sugarapp",
+    "finish-args": [
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--share=ipc",
+        "--device=dri",
```
Comment (TingPing) [inline org.sugarlabs.Pippy.json]:
```suggestion
```

Already have `all`

File: org.sugarlabs.Pippy.json [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+            },
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://cpan.metacpan.org/authors/id/G/GR/GRANTM/XML-Simple-2.25.tar.gz",
+                    "sha256": "531fddaebea2416743eb5c4fdfab028f502123d9a220405a4100e68fc480dbf8"
+                },
+                {
+                    "type": "file",
+                    "path": "perl-MakefilePL-Makefile",
```
Comment (TingPing) [inline org.sugarlabs.Pippy.json]: Might as well just use the `simple` build system type IMO. Not a big deal though.

File: org.sugarlabs.Pippy.json (hunk 1 above)
Comment (tchx84) [reply org.sugarlabs.Pippy.json]: good catch! Thanks.

File: org.sugarlabs.Pippy.json (hunk 2 above)
Comment (tchx84) [reply org.sugarlabs.Pippy.json]: You are right, it's one less file.

After this app, I am planning to do a mass update/cleanup of all of the sugarlabs apps (e.g move most of these modules to a BaseApp). Ill make sure include this, then.

**Comment by tchx84** [comment]
bot, build org.sugarlabs.Pippy

**Comment by nedrichards** [comment]
I get the following error when launching
```
Traceback (most recent call last):
  File "/app/share/sugar/activities/Pippy.activity/pippy_app.py", line 170, in focus
    self._source_tabs.get_text_view().grab_focus()
  File "/app/share/sugar/activities/Pippy.activity/notebook.py", line 320, in get_text_view
    tab = self.get_nth_page(self.get_current_page()).get_children()
AttributeError: 'NoneType' object has no attribute 'get_children'
```
Is that expected?

**Comment by tchx84** [comment]
> I get the following error when launching
> 
> ```
> Traceback (most recent call last):
>   File "/app/share/sugar/activities/Pippy.activity/pippy_app.py", line 170, in focus
>     self._source_tabs.get_text_view().grab_focus()
>   File "/app/share/sugar/activities/Pippy.activity/notebook.py", line 320, in get_text_view
>     tab = self.get_nth_page(self.get_current_page()).get_children()
> AttributeError: 'NoneType' object has no attribute 'get_children'
> ```
> 
> Is that expected? To be clear, other than that, when I opened a new tab/context thing and typed `print("Fun")` Fun was printed ;-)

Nope, it is not. Will look into it.

**Comment by tchx84** [comment]
@nedrichards fixed!

**Comment by tchx84** [comment]
bot, build org.sugarlabs.Pippy

**Comment by nedrichards** [comment]
bot, build org.sugarlabs.Pippy

**Comment by tchx84** [comment]
is there any other blocker ? :)

**Pull request closed without merging**